k9mail-library/src/main/java/com/fsck/k9/mail/internet/ListHeaders.java

```diff
@@ -0,0 +1,46 @@
+package com.fsck.k9.mail.internet;
+
+import android.net.MailTo;
+
+import com.fsck.k9.mail.Address;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Intended to cover:
+ *
+ * RFC 2369
+ * The Use of URLs as Meta-Syntax for Core Mail List Commands
+ * and their Transport through Message Header Fields
+ * https://www.ietf.org/rfc/rfc2369.txt
+ *
+ * This is the following fields:
+ *
+ * List-Help
+ * List-Subscribe
+ * List-Unsubscribe
+ * List-Post
+ * List-Owner
+ * List-Archive
+ *
+ * Currently only provides a utility method for List-Post
+ **/
+public class ListHeaders {
+    public static final String LIST_POST_HEADER = "List-Post";
+    private static final Pattern mailtoContainerPattern = Pattern.compile("<(.+)>");
+
+    public static Address[] parsePostAddress(String headerValue) {
+        if (headerValue == null || headerValue.isEmpty()) {
+            return new Address[]{};
+        } else {
```

Comment:
As a general code style preference I prefer extracting functions to skipping on else statements. It's harder to find all the exit points. You have to read the code inside the if block to know whether it will be affected by the code below. See discussion here: http://programmers.stackexchange.com/a/255528

Some languages actually deprecate the return keyword for this reason.

In terms of nesting - that just means you need to pull out more functions. (Generally for example, if I see a comment demarcating a block of code I look to pull out a function which uses that comment as it's name.)

Anyway I'll refactor to match the code style.